packages/@lightningjs/ui-components/src/docs/ThemingExtensions.mdx

```diff
@@ -145,6 +145,11 @@ If we find through A/B testing, the results are not in favor of the vibrant colo
 
 ## Things to Keep in Mind
 
+### `_construct()` usage restriction
+
+Because extensions are applied at runtime on `_contruct` of the component instance, they cannot properly support a `_construct` method override in the extension so must be avoided.
+Other lifecycle hooks are available and supported, such as `_init` and `_attach`.
```

Comment:
An additional thought here. For components that are created before a theme is applied, functionality added in _init and _attach (lifecycle hooks before _update) would not be run. Any event listener setup, etc. could not be guaranteed to happen. 

Our internal extensions essentially only rely on _update. I wonder if this is almost formalized into a requirement for the extensions classes?

Another option could be that a theme change that has a different set of extensions somehow requires re-initializing any current view of the app. That way we can expect everything to be torn down, and _init/_attach hooks could run with a new extensions applied. 

I think we probably prefer the former at the moment, as live-swapping themes that modify the look of the app seems like the general vision for the feature, even if in practice most will just set a theme once at init time.